File: Tree/Preorder.java
class Node{
    int data;
    Node left, right;

    Node(int data){
        this.data = data;
        left=right=null;
    }
}


public class Preorder {

    static void preorder_traversal(Node root){
        if(root==null) return;

        System.out.print(root.data+" ");
        preorder_traversal(root.left);
        preorder_traversal(root.right);
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);

        preorder_traversal(root);
    }
    
}

File: Tree/Preorder.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

    def preorder_traversal(self, root):
        if root is None:
            return
        print(root.data, end =" ")
        self.preorder_traversal(root.left)
        self.preorder_traversal(root.right)

    def main(self):
        root = Node(1)
        root.left = Node(2)
        root.right = Node(3)

        self.preorder_traversal(root)

if __name__ == "__main__":
    obj = Node(0)
    obj.main()
